snp_blending/urdf/workcell.xacro

```diff
@@ -69,21 +70,25 @@
     <axis xyz="0 0 0"/>
   </joint>-->
 
-  <link name="camera_color_optical_frame"/>
-  <link name="target_cal_frame"/>
+<xacro:property name="camera_cal" value="${load_yaml('$(find snp_automate_2022)/config/camera_calibration.yaml')}" />
 
-  <joint name="camera_cal_joint" type="fixed">
-    <parent link="camera_link"/>
+  <link name="camera_color_optical_frame"/>
+  <joint name="tool0_to_camera" type="fixed">
+    <parent link="tool0"/>
     <child link="camera_color_optical_frame"/>
-    <origin xyz="0.085549560652492 -0.1732631672558103 0.1329373703855808" wxyz="0.5118596229748543 -0.5074314040915237 0.4900425218487801 -0.4902768843111336"/>
+    <origin xyz="${camera_cal['camera_mount_to_camera_pos']['x']} ${camera_cal['camera_mount_to_camera_pos']['y']} ${camera_cal['camera_mount_to_camera_pos']['z']}"
+            rpy="${camera_cal['camera_mount_to_camera_rpy']['x']} ${camera_cal['camera_mount_to_camera_rpy']['y']} ${camera_cal['camera_mount_to_camera_rpy']['z']}" />
   </joint>
 
-  <joint name="target_cal_joint" type="fixed">
-    <parent link="floor"/>
-    <child link="target_cal_frame"/>
-    <origin xyz="0.555641489380775 0.05956001933889868 0.1229851433910979" wxyz="0.6869783604862724 -0.001114251051880176 -0.003909064875724348 -0.7266665052690842"/>
```

Comment:
I would update the `camera_calibration.yaml` file to use these values instead of the ones currently in the file